model/emoji.go

```diff
@@ -40,7 +2426,7 @@ func (emoji *Emoji) IsValid() *AppError {
 		return NewAppError("Emoji.IsValid", "model.emoji.user_id.app_error", nil, "", http.StatusBadRequest)
 	}
 
-	if len(emoji.Name) == 0 || len(emoji.Name) > EMOJI_NAME_MAX_LENGTH || !IsValidAlphaNumHyphenUnderscore(emoji.Name, false) {
+	if len(emoji.Name) == 0 || len(emoji.Name) > EMOJI_NAME_MAX_LENGTH || !IsValidAlphaNumHyphenUnderscore(emoji.Name, false) || StringInMap(systemEmojis, emoji.Name) {
```

Comment:
Instead of having a generic StringInMap function, maybe make an IsSystemEmoji one that only requires the name to be passed in? I think having a general StringInMap function might be misleading since it only works for `map[string]int` and no other map types

Reply:
Sure thing!